### PR TITLE
[CWS] Allow specifying cgroup managers for dumps generation

### PR DESCRIPTION
**Pull request by lebauce**
### What does this PR do?

Allow automatically generating activity dumps for systemd services.

### Motivation

Currently, generating activity dumps for systemd services is supported but it's a manual step.
This PR allows to:
- Automatically generate dumps for systemd services
- Specify the cgroups for which we automatically generate the dumps.

### Describe how you validated your changes

Set `runtime_security_config.activity_dump.cgroup_managers` to `systemd` and check that
activity dumps are only generated for `systemd` cgroups.

### CODE REVIEW

**Review by pgimalac — APPROVED** [review]

File: pkg/security/ebpf/c/include/helpers/activity_dump.h [hunk 1]
```diff
@@ -53,10 +53,16 @@ __attribute__((always_inline)) struct cgroup_tracing_event_t *get_cgroup_tracing
     return evt;
 }
 
+__attribute__((always_inline)) u32 is_cgroup_activity_dumps_supported(struct cgroup_context_t *cgroup) {
+    u32 cgroup_manager = cgroup->cgroup_flags & 0b111;
```
Comment (spikat) [inline pkg/security/ebpf/c/include/helpers/activity_dump.h:57]: could we have a named constant for 0x0b111 ?

File: pkg/security/ebpf/c/include/helpers/activity_dump.h [hunk 2]
```diff
@@ -102,11 +108,15 @@ __attribute__((always_inline)) u64 trace_new_cgroup(void *ctx, u64 now, struct c
         return 0;
     }
 
-    if ((container->cgroup_context.cgroup_flags & 0b111) == CGROUP_MANAGER_SYSTEMD) {
+    if (!is_cgroup_activity_dumps_supported(&container->cgroup_context)) {
         return 0;
     }
 
-    copy_container_id(container->container_id, evt->container.container_id);
+    if ((container->cgroup_context.cgroup_flags&0b111) != CGROUP_MANAGER_SYSTEMD) {
```
Comment (spikat) [inline pkg/security/ebpf/c/include/helpers/activity_dump.h:115]: same here

File: pkg/security/ebpf/c/include/maps.h [hunk 3]
```diff
@@ -28,7 +28,7 @@ BPF_ARRAY_MAP(syscall_ctx_gen_id, u32, 1)
 BPF_ARRAY_MAP(syscall_ctx, char[MAX_SYSCALL_CTX_SIZE], MAX_SYSCALL_CTX_ENTRIES)
 
 BPF_HASH_MAP(activity_dumps_config, u64, struct activity_dump_config, 1) // max entries will be overridden at runtime
-BPF_HASH_MAP(activity_dump_config_defaults, u32, struct activity_dump_config, 1)
+BPF_HASH_MAP(activity_dump_config_defaults, u32, struct activity_dump_config, 4)
```
Comment (spikat) [inline pkg/security/ebpf/c/include/maps.h:31]: If I understand it correctly, shouldn't it be 5 in case systemd is enabled?

File: pkg/security/resolvers/process/resolver_ebpf.go [hunk 4]
```diff
@@ -493,6 +493,11 @@ func (p *EBPFResolver) retrieveExecFileFields(procExecPath string) (*model.FileF
 	return &fileFields, nil
 }
 
+// retrieveExecFileFields fetches inode metadata from kernel space
+func (p *EBPFResolver) retrieveExecFileFields(procExecPath string) (*model.FileFields, error) {
+	return p.retrieveFileFields(procExecPath)
+}
```
Comment (spikat) [inline pkg/security/resolvers/process/resolver_ebpf.go:499]: Not sure to see why this change is needed

File: pkg/security/secl/containerutils/cgroup.go [hunk 5]
```diff
@@ -24,6 +24,12 @@ const (
 	CGroupManagerSystemd                          // systemd
 )
 
+// CGroup flags
+const (
+	SystemdService CGroupFlags = (0 << 8)
```
Comment (spikat) [inline pkg/security/secl/containerutils/cgroup.go:29]: do we agree that's equal to .. zero ? :)

File: pkg/security/config/config.go [hunk 6]
```diff
@@ -368,6 +371,7 @@ func NewRuntimeSecurityConfig() (*RuntimeSecurityConfig, error) {
 		ActivityDumpLoadControlPeriod:         pkgconfigsetup.SystemProbe().GetDuration("runtime_security_config.activity_dump.load_controller_period"),
 		ActivityDumpLoadControlMinDumpTimeout: pkgconfigsetup.SystemProbe().GetDuration("runtime_security_config.activity_dump.min_timeout"),
 		ActivityDumpTracedCgroupsCount:        pkgconfigsetup.SystemProbe().GetInt("runtime_security_config.activity_dump.traced_cgroups_count"),
+		ActivityDumpCgroupsManagers:           pkgconfigsetup.SystemProbe().GetStringSlice("runtime_security_config.activity_dump.cgroup_managers"),
```
Comment (spikat) [inline pkg/security/config/config.go:374]: Shouldn't we add guards/sanitizers around this config?

File: go.mod [hunk 7]
```diff
@@ -151,7 +151,7 @@ require (
 	github.com/DataDog/datadog-agent/pkg/gohai v0.56.0-rc.3
 	github.com/DataDog/datadog-agent/pkg/obfuscate v0.59.0
 	github.com/DataDog/datadog-agent/pkg/remoteconfig/state v0.59.0
-	github.com/DataDog/datadog-agent/pkg/security/secl v0.56.0
```
Comment (paulcacheux) [inline go.mod:154]: Do we really need that ? I think there is a replace anyway

File: pkg/security/security_profile/dump/load_controller.go [hunk 8]
```diff
@@ -58,7 +59,7 @@ func NewActivityDumpLoadController(adm *ActivityDumpManager) (*ActivityDumpLoadC
 	}, nil
 }
 
-func (lc *ActivityDumpLoadController) getDefaultLoadConfig() *model.ActivityDumpLoadConfig {
+func (lc *ActivityDumpLoadController) getDefaultLoadConfigs() (map[containerutils.CGroupManager]*model.ActivityDumpLoadConfig, error) {
```
Comment (spikat) [inline pkg/security/security_profile/dump/load_controller.go:62]: As this func is called multiple times in case of silent workloads, could we initialize only once the default configs?

File: pkg/security/secl/containerutils/cgroup.go (hunk 5 above)
Comment (lebauce) [reply pkg/security/secl/containerutils/cgroup.go:29]: Yes :-)

File: pkg/security/config/config.go (hunk 6 above)
Comment (lebauce) [reply pkg/security/config/config.go:374]: You mean if we specify a non-existing cgroup manager ? I would say it's handled in pkg/security/security_profile/dump/load_controller.go

File: pkg/security/security_profile/dump/load_controller.go (hunk 8 above)
Comment (lebauce) [reply pkg/security/security_profile/dump/load_controller.go:62]: Done

File: pkg/security/resolvers/process/resolver_ebpf.go (hunk 4 above)
Comment (lebauce) [reply pkg/security/resolvers/process/resolver_ebpf.go:499]: It's not anymore, indeed

File: pkg/security/ebpf/c/include/maps.h (hunk 3 above)
Comment (lebauce) [reply pkg/security/ebpf/c/include/maps.h:31]: Good catch, I did a last minute test and committed it by mistake

File: pkg/security/ebpf/c/include/helpers/activity_dump.h (hunk 2 above)
Comment (lebauce) [reply pkg/security/ebpf/c/include/helpers/activity_dump.h:115]: Done

File: pkg/security/ebpf/c/include/helpers/activity_dump.h (hunk 1 above)
Comment (lebauce) [reply pkg/security/ebpf/c/include/helpers/activity_dump.h:57]: Done

Comment (spikat) [reply pkg/security/ebpf/c/include/helpers/activity_dump.h:57]: Thanks!

File: go.mod (hunk 7 above)
Comment (lebauce) [reply go.mod:154]: :+1:

File: pkg/security/config/config.go (hunk 6 above)
Comment (spikat) [reply pkg/security/config/config.go:374]: Exact!

**Review by spikat — APPROVED** [review]
